tests/unit/test_serialisation/test_serialisation.py

```diff
@@ -591,3 +594,339 @@ def test_serialised_model_plotting(self):
+        assert int_var_json["name"] == "x"
+
+        expr2 = Serialise.convert_symbol_from_json(json_dict)
+        assert isinstance(expr2, pybamm.IndefiniteIntegral)
+        assert isinstance(expr2.child, pybamm.SpatialVariable)
+
+        assert expr2.child.name == "x"
+        assert isinstance(expr2.integration_variable, list)
+        assert len(expr2.integration_variable) == 1
+        assert isinstance(expr2.integration_variable[0], pybamm.SpatialVariable)
+        assert expr2.integration_variable[0].name == "x"
+
+    def test_symbol_fallback_serialisation(self):
+        var = pybamm.Variable("v", domain="electrode")
+        diff = pybamm.Gradient(var)
+        json_dict = Serialise.convert_symbol_to_json(diff)
+        diff2 = Serialise.convert_symbol_from_json(json_dict)
+
+        assert isinstance(diff2, pybamm.Gradient)
+        assert isinstance(diff2.children[0], pybamm.Variable)
+        assert diff2.children[0].name == "v"
+        assert diff2.children[0].domains["primary"] == ["electrode"]
+
+    def test_unhandled_symbol_type_error(self):
+        class NotSymbol:
+            def __init__(self):
+                self.name = "not_a_symbol"
+
+        dummy = NotSymbol()
+        with pytest.raises(ValueError) as e:
+            Serialise.convert_symbol_to_json(dummy)
+
+        assert "Error processing 'not_a_symbol'. Unknown symbol type:" in str(e.value)
+
+    def test_deserialising_unhandled_type(self):
+        unhandled_json = {"type": "NotARealSymbol", "foo": "bar"}
+        with pytest.raises(
+            ValueError,
+            match=r"Unhandled symbol type or malformed entry: .*NotARealSymbol",
+        ):
+            Serialise.convert_symbol_from_json(unhandled_json)
+
+        unhandled_json2 = {"a": 1, "b": 2}
+        with pytest.raises(
+            ValueError, match=r"Unhandled symbol type or malformed entry: .*"
+        ):
+            Serialise.convert_symbol_from_json(unhandled_json2)
+
+    def test_save_and_load_custom_model(self):
+        model = pybamm.BaseModel(name="test_model")
+        a = pybamm.Variable("a", domain="electrode")
+        b = pybamm.Variable("b", domain="electrode")
+        model.rhs = {a: b}
+        model.initial_conditions = {a: pybamm.Scalar(1)}
+        model.algebraic = {}
+        model.boundary_conditions = {a: {"left": (pybamm.Scalar(0), "Dirichlet")}}
+        model.events = [pybamm.Event("terminal", pybamm.Scalar(1) - b, "TERMINATION")]
+        model.variables = {"a": a, "b": b}
+
+        # save model
+        Serialise.save_custom_model(model, filename="test_model")
+
+        # check json exists
+        assert os.path.exists("test_model.json")
+
+        # saving with defualt filename
+        Serialise().save_custom_model(model)
+        pattern = r"test_model_\d{4}_\d{2}_\d{2}-(AM|PM)\d{2}_\d{2}\.json"
+        matched = [f for f in os.listdir(".") if re.fullmatch(pattern, f)]
+        assert matched
```

Comment:
I think this should be:

```suggestion
        # saving with default filename
        Serialise().save_custom_model(model)
        pattern = r"test_model_\d{4}_\d{2}_\d{2}_\d{2}_\d{2}_\d{2}\.json"
        matched = [f for f in os.listdir(".") if re.fullmatch(pattern, f)]
        assert matched
```